Output processor is not run if versionCode is already bigger than biggest uploaded

### Describe the bug

Exception while publishing APK to Google play

```
FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:publishReleaseApk'.
> Concurrent uploads for variant release. Make sure to synchronously upload your APKs such that they don't conflict.

* Try:
Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Exception is:
org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':app:publishReleaseApk'.
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:110)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.execute(ExecuteActionsTaskExecuter.java:77)
        at org.gradle.api.internal.tasks.execution.OutputDirectoryCreatingTaskExecuter.execute(OutputDirectoryCreatingTaskExecuter.java:51)
        at org.gradle.api.internal.tasks.execution.SkipUpToDateTaskExecuter.execute(SkipUpToDateTaskExecuter.java:59)
        at org.gradle.api.internal.tasks.execution.ResolveTaskOutputCachingStateExecuter.execute(ResolveTaskOutputCachingStateExecuter.java:54)
        at org.gradle.api.internal.tasks.execution.ResolveBuildCacheKeyExecuter.execute(ResolveBuildCacheKeyExecuter.java:79)
        at org.gradle.api.internal.tasks.execution.ValidatingTaskExecuter.execute(ValidatingTaskExecuter.java:59)
        at org.gradle.api.internal.tasks.execution.SkipEmptySourceFilesTaskExecuter.execute(SkipEmptySourceFilesTaskExecuter.java:101)
        at org.gradle.api.internal.tasks.execution.FinalizeInputFilePropertiesTaskExecuter.execute(FinalizeInputFilePropertiesTaskExecuter.java:44)
        at org.gradle.api.internal.tasks.execution.CleanupStaleOutputsExecuter.execute(CleanupStaleOutputsExecuter.java:91)
        at org.gradle.api.internal.tasks.execution.ResolveTaskArtifactStateTaskExecuter.execute(ResolveTaskArtifactStateTaskExecuter.java:62)
        at org.gradle.api.internal.tasks.execution.SkipTaskWithNoActionsExecuter.execute(SkipTaskWithNoActionsExecuter.java:59)
        at org.gradle.api.internal.tasks.execution.SkipOnlyIfTaskExecuter.execute(SkipOnlyIfTaskExecuter.java:54)
        at org.gradle.api.internal.tasks.execution.ExecuteAtMostOnceTaskExecuter.execute(ExecuteAtMostOnceTaskExecuter.java:43)
        at org.gradle.api.internal.tasks.execution.CatchExceptionTaskExecuter.execute(CatchExceptionTaskExecuter.java:34)
        at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.run(EventFiringTaskExecuter.java:51)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:300)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:292)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:174)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:90)
        at org.gradle.internal.operations.DelegatingBuildOperationExecutor.run(DelegatingBuildOperationExecutor.java:31)
        at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter.execute(EventFiringTaskExecuter.java:46)
        at org.gradle.execution.taskgraph.LocalTaskInfoExecutor.execute(LocalTaskInfoExecutor.java:42)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareWorkItemExecutor.execute(DefaultTaskExecutionGraph.java:277)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareWorkItemExecutor.execute(DefaultTaskExecutionGraph.java:262)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$ExecutorWorker$1.execute(DefaultTaskPlanExecutor.java:135)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$ExecutorWorker$1.execute(DefaultTaskPlanExecutor.java:130)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$ExecutorWorker.execute(DefaultTaskPlanExecutor.java:200)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$ExecutorWorker.executeWithWork(DefaultTaskPlanExecutor.java:191)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$ExecutorWorker.run(DefaultTaskPlanExecutor.java:130)
        at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:63)
        at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:46)
        at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:55)
Caused by: java.lang.IllegalStateException: Concurrent uploads for variant release. Make sure to synchronously upload your APKs such that they don't conflict.
        at com.github.triplet.gradle.play.tasks.internal.PlayPublishPackageBase.handleUploadFailures(PlayPublishPackageBase.kt:128)
        at com.github.triplet.gradle.play.tasks.PublishApk.publishApk(PublishApk.kt:65)
        at com.github.triplet.gradle.play.tasks.PublishApk.access$publishApk(PublishApk.kt:21)
        at com.github.triplet.gradle.play.tasks.PublishApk$publishApks$1$1.execute(PublishApk.kt:47)
        at com.github.triplet.gradle.play.tasks.PublishApk$publishApks$1$1.execute(PublishApk.kt:21)
        at org.gradle.api.internal.changedetection.changes.RebuildIncrementalTaskInputs.doOutOfDate(RebuildIncrementalTaskInputs.java:45)
        at org.gradle.api.internal.changedetection.changes.StatefulIncrementalTaskInputs.outOfDate(StatefulIncrementalTaskInputs.java:32)
        at org.gradle.api.internal.changedetection.changes.RebuildIncrementalTaskInputs.outOfDate(RebuildIncrementalTaskInputs.java:29)
        at com.github.triplet.gradle.play.tasks.PublishApk$publishApks$1.invoke(PublishApk.kt:44)
        at com.github.triplet.gradle.play.tasks.PublishApk$publishApks$1.invoke(PublishApk.kt:21)
        at com.github.triplet.gradle.play.tasks.internal.PlayPublishTaskBase$write$1.invoke(PlayPublishTaskBase.kt:76)
        at com.github.triplet.gradle.play.tasks.internal.PlayPublishTaskBase$write$1.invoke(PlayPublishTaskBase.kt:20)
        at com.github.triplet.gradle.play.tasks.internal.PlayPublishTaskBase.read(PlayPublishTaskBase.kt:72)
        at com.github.triplet.gradle.play.tasks.internal.PlayPublishTaskBase.read$default(PlayPublishTaskBase.kt:39)
        at com.github.triplet.gradle.play.tasks.internal.PlayPublishTaskBase.write(PlayPublishTaskBase.kt:75)
        at com.github.triplet.gradle.play.tasks.PublishApk.publishApks(PublishApk.kt:38)
        at org.gradle.internal.reflect.JavaMethod.invoke(JavaMethod.java:73)
        at org.gradle.api.internal.project.taskfactory.IncrementalTaskAction.doExecute(IncrementalTaskAction.java:50)
        at org.gradle.api.internal.project.taskfactory.StandardTaskAction.execute(StandardTaskAction.java:39)
        at org.gradle.api.internal.project.taskfactory.StandardTaskAction.execute(StandardTaskAction.java:26)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter$1.run(ExecuteActionsTaskExecuter.java:131)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:300)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:292)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:174)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:90)
        at org.gradle.internal.operations.DelegatingBuildOperationExecutor.run(DelegatingBuildOperationExecutor.java:31)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeAction(ExecuteActionsTaskExecuter.java:120)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:99)
        ... 32 more
Caused by: com.google.api.client.googleapis.json.GoogleJsonResponseException: 403 Forbidden
{
  "code" : 403,
  "errors" : [ {
    "domain" : "androidpublisher",
    "message" : "APK specifies a version code that has already been used.",
    "reason" : "apkUpgradeVersionConflict"
  } ],
  "message" : "APK specifies a version code that has already been used."
}
        at com.google.api.client.googleapis.json.GoogleJsonResponseException.from(GoogleJsonResponseException.java:150)
        at com.google.api.client.googleapis.services.json.AbstractGoogleJsonClientRequest.newExceptionOnError(AbstractGoogleJsonClientRequest.java:113)
        at com.google.api.client.googleapis.services.json.AbstractGoogleJsonClientRequest.newExceptionOnError(AbstractGoogleJsonClientRequest.java:40)
        at com.google.api.client.googleapis.services.AbstractGoogleClientRequest.executeUnparsed(AbstractGoogleClientRequest.java:432)
        at com.google.api.client.googleapis.services.AbstractGoogleClientRequest.executeUnparsed(AbstractGoogleClientRequest.java:352)
        at com.google.api.client.googleapis.services.AbstractGoogleClientRequest.execute(AbstractGoogleClientRequest.java:469)
        at com.github.triplet.gradle.play.tasks.PublishApk.publishApk(PublishApk.kt:63)
        ... 58 more

```

Configuration is
```groovy
   play {
        serviceAccountCredentials = file("${home}/play/Google Play Android Developer.json")
        track = "internal"
        releaseStatus = "draft"
        resolutionStrategy = "auto"
        outputProcessor {
            versionNameOverride = "$versionNameOverride.$versionCode"
        }
    }
```
#### Versions

 - Gradle Play Publisher: 2.1.0
 - Gradle Wrapper (if applicable): 4.6
-  Gradle: gradle-4.10.1-all.zip 
 - Android Gradle Plugin (if applicable): 3.3.2

Hmm... No. Below the complete log of the last build:
```
simple	06-Mar-2019 19:58:25	Build Groove Studio - Dance Classes - Groove Studio (Android) - Default Job #16 (GSDC-GSA-JOB1-16) started building on agent Default Agent
simple	06-Mar-2019 19:58:25	
simple	06-Mar-2019 19:58:25	Build working directory is /var/atlassian/application-data/bamboo/xml-data/build-dir/GSDC-GSA-JOB1
command	06-Mar-2019 19:58:25	Substituting variable: ${bamboo.working.directory} with /var/atlassian/application-data/bamboo/xml-data/build-dir/GSDC-GSA-JOB1
command	06-Mar-2019 19:58:25	Substituting variable: ${bamboo.working.directory} with /var/atlassian/application-data/bamboo/xml-data/build-dir/GSDC-GSA-JOB1
command	06-Mar-2019 19:58:25	Substituting variable: ${bamboo.tmp.directory} with /var/atlassian/application-data/bamboo/temp
command	06-Mar-2019 19:58:25	Substituting variable: ${bamboo.tmp.directory} with /var/atlassian/application-data/bamboo/temp
command	06-Mar-2019 19:58:25	Beginning to execute external process for build 'Groove Studio - Dance Classes - Groove Studio (Android) - Default Job #16 (GSDC-GSA-JOB1-16)'\n ... running command line: \n/usr/bin/docker run --volume /var/atlassian/application-data/bamboo/xml-data/build-dir/GSDC-GSA-JOB1:/var/atlassian/application-data/bamboo/xml-data/build-dir/GSDC-GSA-JOB1 --volume /var/atlassian/application-data/bamboo/temp:/var/atlassian/application-data/bamboo/temp --volume /home/bamboo/.gradle/:/root/.gradle/ --detach --name 914eb96b-c4ea-4f32-9430-c83bcb14307058327056 --net=host mingc/android-build-box:latest tail -f /dev/null\n ... in: /var/atlassian/application-data/bamboo/xml-data/build-dir/GSDC-GSA-JOB1\n
build	06-Mar-2019 19:58:25	b4ef342306076c194f9a479dc37b403ab82976f571627b96c9718cfe118a31ab
command	06-Mar-2019 19:58:27	Beginning to execute external process for build 'Groove Studio - Dance Classes - Groove Studio (Android) - Default Job #16 (GSDC-GSA-JOB1-16)'\n ... running command line: \n/usr/bin/docker cp /opt/atlassian/bamboo/temp/initialiseDockerContainer.sh5264701185191267650.tmp 914eb96b-c4ea-4f32-9430-c83bcb14307058327056:/tmp/initialiseContainer.sh\n ... in: /var/atlassian/application-data/bamboo/xml-data/build-dir/GSDC-GSA-JOB1\n
command	06-Mar-2019 19:58:27	Beginning to execute external process for build 'Groove Studio - Dance Classes - Groove Studio (Android) - Default Job #16 (GSDC-GSA-JOB1-16)'\n ... running command line: \n/usr/bin/docker exec -u root 914eb96b-c4ea-4f32-9430-c83bcb14307058327056 chown root:root /tmp/initialiseContainer.sh\n ... in: /var/atlassian/application-data/bamboo/xml-data/build-dir/GSDC-GSA-JOB1\n
command	06-Mar-2019 19:58:27	Beginning to execute external process for build 'Groove Studio - Dance Classes - Groove Studio (Android) - Default Job #16 (GSDC-GSA-JOB1-16)'\n ... running command line: \n/usr/bin/docker exec -u root 914eb96b-c4ea-4f32-9430-c83bcb14307058327056 chmod 755 /tmp/initialiseContainer.sh\n ... in: /var/atlassian/application-data/bamboo/xml-data/build-dir/GSDC-GSA-JOB1\n
command	06-Mar-2019 19:58:27	Beginning to execute external process for build 'Groove Studio - Dance Classes - Groove Studio (Android) - Default Job #16 (GSDC-GSA-JOB1-16)'\n ... running command line: \n/usr/bin/docker exec 914eb96b-c4ea-4f32-9430-c83bcb14307058327056 /tmp/initialiseContainer.sh\n ... in: /var/atlassian/application-data/bamboo/xml-data/build-dir/GSDC-GSA-JOB1\n
simple	06-Mar-2019 19:58:28	Executing build Groove Studio - Dance Classes - Groove Studio (Android) - Default Job #16 (GSDC-GSA-JOB1-16)
simple	06-Mar-2019 19:58:28	Starting task 'Checkout Default Repository' of type 'com.atlassian.bamboo.plugins.vcs:task.vcs.checkout'
simple	06-Mar-2019 19:58:28	Checking out into /var/atlassian/application-data/bamboo/xml-data/build-dir/GSDC-GSA-JOB1
simple	06-Mar-2019 19:58:28	Updating source code to revision: 5d84c755165dc412fd51b88de321496cc9053d14
simple	06-Mar-2019 19:58:28	Retrieving source from remote repository...
simple	06-Mar-2019 19:58:28	Pulling from repository [https://build:********@bitbucket.org/cathy/groovestudio] to directory '/var/atlassian/application-data/bamboo/xml-data/build-dir/_hg-repositories-cache/08b2ff34284d85ddff06779fd7d9a36b41853adf' revision `5d84c755165dc412fd51b88de321496cc9053d14'
simple	06-Mar-2019 19:58:29	pulling from https://build:***@bitbucket.org/cathy/groovestudio
simple	06-Mar-2019 19:58:29	no changes found
simple	06-Mar-2019 19:58:29	Obtaining remote repository to agent's cache directory '/var/atlassian/application-data/bamboo/xml-data/build-dir/_hg-repositories-cache/08b2ff34284d85ddff06779fd7d9a36b41853adf'...
simple	06-Mar-2019 19:58:29	Pulling from repository [/var/atlassian/application-data/bamboo/xml-data/build-dir/_hg-repositories-cache/08b2ff34284d85ddff06779fd7d9a36b41853adf] to directory '/var/atlassian/application-data/bamboo/xml-data/build-dir/GSDC-GSA-JOB1' revision `5d84c755165dc412fd51b88de321496cc9053d14'
simple	06-Mar-2019 19:58:29	pulling from /var/atlassian/application-data/bamboo/xml-data/build-dir/_hg-repositories-cache/08b2ff34284d85ddff06779fd7d9a36b41853adf
simple	06-Mar-2019 19:58:29	no changes found
simple	06-Mar-2019 19:58:29	1 files updated, 0 files merged, 0 files removed, 0 files unresolved
simple	06-Mar-2019 19:58:29	Updated source code to revision: 5d84c755165dc412fd51b88de321496cc9053d14
simple	06-Mar-2019 19:58:29	Finished task 'Checkout Default Repository' with result: Success
simple	06-Mar-2019 19:58:29	Running pre-build action: VCS Version Collector
simple	06-Mar-2019 19:58:29	Running pre-build action: Sauce OnDemand Build Configurator
simple	06-Mar-2019 19:58:30	Starting task 'Build' of type 'com.atlassian.bamboo.plugins.scripttask:task.builder.script'
command	06-Mar-2019 19:58:30	Beginning to execute external process for build 'Groove Studio - Dance Classes - Groove Studio (Android) - Default Job #16 (GSDC-GSA-JOB1-16)'\n ... running command line: \n/opt/atlassian/bamboo/temp/runInDocker494432707447173389.sh /var/atlassian/application-data/bamboo/temp/GSDC-GSA-JOB1-16-ScriptBuildTask-97456339034608373.sh\n ... in: /opt/atlassian/bamboo/temp\n
build	06-Mar-2019 19:58:31	Starting a Gradle Daemon, 1 incompatible and 3 stopped Daemons could not be reused, use --status for details
build	06-Mar-2019 19:58:34	
build	06-Mar-2019 19:58:34	> Configure project :
build	06-Mar-2019 19:58:34	CI is true
build	06-Mar-2019 19:58:36	
build	06-Mar-2019 19:58:36	> Configure project :app
build	06-Mar-2019 19:58:36	inside play CI is true
build	06-Mar-2019 19:58:36	registerResGeneratingTask is deprecated, use registerGeneratedResFolders(FileCollection)
build	06-Mar-2019 19:58:36	registerResGeneratingTask is deprecated, use registerGeneratedResFolders(FileCollection)
build	06-Mar-2019 19:58:36	WARNING: API 'variantOutput.getPackageApplication()' is obsolete and has been replaced with 'variant.getPackageApplicationProvider()'.
build	06-Mar-2019 19:58:36	It will be removed at the end of 2019.
build	06-Mar-2019 19:58:36	For more information, see https://d.android.com/r/tools/task-configuration-avoidance.
build	06-Mar-2019 19:58:36	To determine what is calling variantOutput.getPackageApplication(), use -Pandroid.debug.obsoleteApi=true on the command line to display a stack trace.
build	06-Mar-2019 19:58:37	
build	06-Mar-2019 19:58:37	> Task :app:preBuild UP-TO-DATE
build	06-Mar-2019 19:58:37	> Task :app:extractProguardFiles UP-TO-DATE
build	06-Mar-2019 19:58:37	> Task :app:preReleaseBuild UP-TO-DATE
build	06-Mar-2019 19:58:37	> Task :app:compileReleaseAidl NO-SOURCE
build	06-Mar-2019 19:58:37	> Task :app:compileReleaseRenderscript UP-TO-DATE
build	06-Mar-2019 19:58:40	> Task :app:processReleaseMetadata
build	06-Mar-2019 19:58:40	> Task :app:checkReleaseManifest UP-TO-DATE
build	06-Mar-2019 19:58:40	> Task :app:generateReleaseBuildConfig UP-TO-DATE
build	06-Mar-2019 19:58:40	> Task :app:prepareLintJar UP-TO-DATE
build	06-Mar-2019 19:58:40	> Task :app:writeReleaseApplicationId UP-TO-DATE
build	06-Mar-2019 19:58:40	
build	06-Mar-2019 19:58:40	> Task :app:processReleaseGoogleServices
build	06-Mar-2019 19:58:40	Parsing json file: /var/atlassian/application-data/bamboo/xml-data/build-dir/GSDC-GSA-JOB1/app/google-services.json
build	06-Mar-2019 19:58:40	
build	06-Mar-2019 19:58:40	> Task :app:generateReleaseSources
build	06-Mar-2019 19:58:40	> Task :app:javaPreCompileRelease UP-TO-DATE
build	06-Mar-2019 19:58:40	> Task :app:mainApkListPersistenceRelease UP-TO-DATE
build	06-Mar-2019 19:58:40	> Task :app:generateReleaseResValues UP-TO-DATE
build	06-Mar-2019 19:58:40	> Task :app:generateReleaseResources UP-TO-DATE
build	06-Mar-2019 19:58:40	> Task :app:mergeReleaseResources UP-TO-DATE
build	06-Mar-2019 19:58:40	> Task :app:createReleaseCompatibleScreenManifests UP-TO-DATE
build	06-Mar-2019 19:58:40	> Task :app:processReleaseManifest UP-TO-DATE
build	06-Mar-2019 19:58:40	> Task :app:processReleaseResources UP-TO-DATE
build	06-Mar-2019 19:58:40	> Task :app:compileReleaseJavaWithJavac UP-TO-DATE
build	06-Mar-2019 19:58:40	> Task :app:compileReleaseNdk NO-SOURCE
build	06-Mar-2019 19:58:40	> Task :app:compileReleaseSources UP-TO-DATE
build	06-Mar-2019 19:58:42	> Task :app:lintVitalRelease
build	06-Mar-2019 19:58:42	> Task :app:mergeReleaseShaders UP-TO-DATE
build	06-Mar-2019 19:58:42	> Task :app:compileReleaseShaders UP-TO-DATE
build	06-Mar-2019 19:58:42	> Task :app:generateReleaseAssets UP-TO-DATE
build	06-Mar-2019 19:58:42	> Task :app:mergeReleaseAssets UP-TO-DATE
build	06-Mar-2019 19:58:42	> Task :app:validateSigningRelease UP-TO-DATE
build	06-Mar-2019 19:58:42	> Task :app:signingConfigWriterRelease UP-TO-DATE
build	06-Mar-2019 19:58:42	> Task :app:processReleaseJavaRes NO-SOURCE
build	06-Mar-2019 19:58:42	> Task :app:transformResourcesWithMergeJavaResForRelease UP-TO-DATE
build	06-Mar-2019 19:58:42	> Task :app:transformClassesAndResourcesWithProguardForRelease UP-TO-DATE
build	06-Mar-2019 19:58:42	> Task :app:transformClassesWithDexBuilderForRelease UP-TO-DATE
build	06-Mar-2019 19:58:42	> Task :app:transformDexArchiveWithDexMergerForRelease UP-TO-DATE
build	06-Mar-2019 19:58:42	> Task :app:mergeReleaseJniLibFolders UP-TO-DATE
build	06-Mar-2019 19:58:42	> Task :app:transformNativeLibsWithMergeJniLibsForRelease UP-TO-DATE
build	06-Mar-2019 19:58:42	> Task :app:transformNativeLibsWithStripDebugSymbolForRelease UP-TO-DATE
build	06-Mar-2019 19:58:42	> Task :app:packageRelease UP-TO-DATE
build	06-Mar-2019 19:58:42	> Task :app:assembleRelease
build	06-Mar-2019 19:58:42	> Task :app:generateReleasePlayResources UP-TO-DATE
build	06-Mar-2019 19:59:12	> Task :app:publishReleaseApk
build	06-Mar-2019 20:00:22	> Task :app:publishReleaseApk FAILED
error	06-Mar-2019 20:00:22	
error	06-Mar-2019 20:00:22	FAILURE: Build failed with an exception.
error	06-Mar-2019 20:00:22	
error	06-Mar-2019 20:00:22	* What went wrong:
error	06-Mar-2019 20:00:22	Execution failed for task ':app:publishReleaseApk'.
error	06-Mar-2019 20:00:22	> Concurrent uploads for variant release. Make sure to synchronously upload your APKs such that they don't conflict.
error	06-Mar-2019 20:00:22	
error	06-Mar-2019 20:00:22	* Try:
error	06-Mar-2019 20:00:22	Run with --info or --debug option to get more log output. Run with --scan to get full insights.
error	06-Mar-2019 20:00:22	
error	06-Mar-2019 20:00:22	* Exception is:
error	06-Mar-2019 20:00:22	org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':app:publishReleaseApk'.
error	06-Mar-2019 20:00:22		at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:110)
error	06-Mar-2019 20:00:22		at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.execute(ExecuteActionsTaskExecuter.java:77)
error	06-Mar-2019 20:00:22		at org.gradle.api.internal.tasks.execution.OutputDirectoryCreatingTaskExecuter.execute(OutputDirectoryCreatingTaskExecuter.java:51)
error	06-Mar-2019 20:00:22		at org.gradle.api.internal.tasks.execution.SkipUpToDateTaskExecuter.execute(SkipUpToDateTaskExecuter.java:59)
error	06-Mar-2019 20:00:22		at org.gradle.api.internal.tasks.execution.ResolveTaskOutputCachingStateExecuter.execute(ResolveTaskOutputCachingStateExecuter.java:54)
error	06-Mar-2019 20:00:22		at org.gradle.api.internal.tasks.execution.ResolveBuildCacheKeyExecuter.execute(ResolveBuildCacheKeyExecuter.java:79)
error	06-Mar-2019 20:00:22		at org.gradle.api.internal.tasks.execution.ValidatingTaskExecuter.execute(ValidatingTaskExecuter.java:59)
error	06-Mar-2019 20:00:22		at org.gradle.api.internal.tasks.execution.SkipEmptySourceFilesTaskExecuter.execute(SkipEmptySourceFilesTaskExecuter.java:101)
error	06-Mar-2019 20:00:22		at org.gradle.api.internal.tasks.execution.FinalizeInputFilePropertiesTaskExecuter.execute(FinalizeInputFilePropertiesTaskExecuter.java:44)
error	06-Mar-2019 20:00:22		at org.gradle.api.internal.tasks.execution.CleanupStaleOutputsExecuter.execute(CleanupStaleOutputsExecuter.java:91)
error	06-Mar-2019 20:00:22		at org.gradle.api.internal.tasks.execution.ResolveTaskArtifactStateTaskExecuter.execute(ResolveTaskArtifactStateTaskExecuter.java:62)
error	06-Mar-2019 20:00:22		at org.gradle.api.internal.tasks.execution.SkipTaskWithNoActionsExecuter.execute(SkipTaskWithNoActionsExecuter.java:59)
error	06-Mar-2019 20:00:22		at org.gradle.api.internal.tasks.execution.SkipOnlyIfTaskExecuter.execute(SkipOnlyIfTaskExecuter.java:54)
error	06-Mar-2019 20:00:22		at org.gradle.api.internal.tasks.execution.ExecuteAtMostOnceTaskExecuter.execute(ExecuteAtMostOnceTaskExecuter.java:43)
error	06-Mar-2019 20:00:22		at org.gradle.api.internal.tasks.execution.CatchExceptionTaskExecuter.execute(CatchExceptionTaskExecuter.java:34)
error	06-Mar-2019 20:00:22		at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.run(EventFiringTaskExecuter.java:51)
error	06-Mar-2019 20:00:22		at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:300)
error	06-Mar-2019 20:00:22		at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:292)
error	06-Mar-2019 20:00:22		at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:174)
error	06-Mar-2019 20:00:22		at org.gradle.internal.operations.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:90)
error	06-Mar-2019 20:00:22		at org.gradle.internal.operations.DelegatingBuildOperationExecutor.run(DelegatingBuildOperationExecutor.java:31)
error	06-Mar-2019 20:00:22		at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter.execute(EventFiringTaskExecuter.java:46)
error	06-Mar-2019 20:00:22		at org.gradle.execution.taskgraph.LocalTaskInfoExecutor.execute(LocalTaskInfoExecutor.java:42)
error	06-Mar-2019 20:00:22		at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareWorkItemExecutor.execute(DefaultTaskExecutionGraph.java:277)
error	06-Mar-2019 20:00:22		at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareWorkItemExecutor.execute(DefaultTaskExecutionGraph.java:262)
error	06-Mar-2019 20:00:22		at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$ExecutorWorker$1.execute(DefaultTaskPlanExecutor.java:135)
error	06-Mar-2019 20:00:22		at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$ExecutorWorker$1.execute(DefaultTaskPlanExecutor.java:130)
error	06-Mar-2019 20:00:22		at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$ExecutorWorker.execute(DefaultTaskPlanExecutor.java:200)
error	06-Mar-2019 20:00:22		at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$ExecutorWorker.executeWithWork(DefaultTaskPlanExecutor.java:191)
error	06-Mar-2019 20:00:22		at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$ExecutorWorker.run(DefaultTaskPlanExecutor.java:130)
error	06-Mar-2019 20:00:22		at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:63)
error	06-Mar-2019 20:00:22		at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:46)
error	06-Mar-2019 20:00:22		at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:55)
error	06-Mar-2019 20:00:22	Caused by: java.lang.IllegalStateException: Concurrent uploads for variant release. Make sure to synchronously upload your APKs such that they don't conflict.
error	06-Mar-2019 20:00:22		at com.github.triplet.gradle.play.tasks.internal.PlayPublishPackageBase.handleUploadFailures(PlayPublishPackageBase.kt:128)
error	06-Mar-2019 20:00:22		at com.github.triplet.gradle.play.tasks.PublishApk.publishApk(PublishApk.kt:65)
error	06-Mar-2019 20:00:22		at com.github.triplet.gradle.play.tasks.PublishApk.access$publishApk(PublishApk.kt:21)
error	06-Mar-2019 20:00:22		at com.github.triplet.gradle.play.tasks.PublishApk$publishApks$1$1.execute(PublishApk.kt:47)
error	06-Mar-2019 20:00:22		at com.github.triplet.gradle.play.tasks.PublishApk$publishApks$1$1.execute(PublishApk.kt:21)
error	06-Mar-2019 20:00:22		at org.gradle.api.internal.changedetection.changes.RebuildIncrementalTaskInputs.doOutOfDate(RebuildIncrementalTaskInputs.java:45)
error	06-Mar-2019 20:00:22		at org.gradle.api.internal.changedetection.changes.StatefulIncrementalTaskInputs.outOfDate(StatefulIncrementalTaskInputs.java:32)
error	06-Mar-2019 20:00:22		at org.gradle.api.internal.changedetection.changes.RebuildIncrementalTaskInputs.outOfDate(RebuildIncrementalTaskInputs.java:29)
error	06-Mar-2019 20:00:22		at com.github.triplet.gradle.play.tasks.PublishApk$publishApks$1.invoke(PublishApk.kt:44)
error	06-Mar-2019 20:00:22		at com.github.triplet.gradle.play.tasks.PublishApk$publishApks$1.invoke(PublishApk.kt:21)
error	06-Mar-2019 20:00:22		at com.github.triplet.gradle.play.tasks.internal.PlayPublishTaskBase$write$1.invoke(PlayPublishTaskBase.kt:76)
error	06-Mar-2019 20:00:22		at com.github.triplet.gradle.play.tasks.internal.PlayPublishTaskBase$write$1.invoke(PlayPublishTaskBase.kt:20)
error	06-Mar-2019 20:00:22		at com.github.triplet.gradle.play.tasks.internal.PlayPublishTaskBase.read(PlayPublishTaskBase.kt:72)
error	06-Mar-2019 20:00:22		at com.github.triplet.gradle.play.tasks.internal.PlayPublishTaskBase.read$default(PlayPublishTaskBase.kt:39)
error	06-Mar-2019 20:00:22		at com.github.triplet.gradle.play.tasks.internal.PlayPublishTaskBase.write(PlayPublishTaskBase.kt:75)
error	06-Mar-2019 20:00:22		at com.github.triplet.gradle.play.tasks.PublishApk.publishApks(PublishApk.kt:38)
error	06-Mar-2019 20:00:22		at org.gradle.internal.reflect.JavaMethod.invoke(JavaMethod.java:73)
error	06-Mar-2019 20:00:22		at org.gradle.api.internal.project.taskfactory.IncrementalTaskAction.doExecute(IncrementalTaskAction.java:50)
error	06-Mar-2019 20:00:22		at org.gradle.api.internal.project.taskfactory.StandardTaskAction.execute(StandardTaskAction.java:39)
error	06-Mar-2019 20:00:22		at org.gradle.api.internal.project.taskfactory.StandardTaskAction.execute(StandardTaskAction.java:26)
error	06-Mar-2019 20:00:22		at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter$1.run(ExecuteActionsTaskExecuter.java:131)
error	06-Mar-2019 20:00:22		at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:300)
error	06-Mar-2019 20:00:22		at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:292)
error	06-Mar-2019 20:00:22		at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:174)
error	06-Mar-2019 20:00:22		at org.gradle.internal.operations.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:90)
error	06-Mar-2019 20:00:22		at org.gradle.internal.operations.DelegatingBuildOperationExecutor.run(DelegatingBuildOperationExecutor.java:31)
error	06-Mar-2019 20:00:22		at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeAction(ExecuteActionsTaskExecuter.java:120)
error	06-Mar-2019 20:00:22		at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:99)
error	06-Mar-2019 20:00:22		... 32 more
error	06-Mar-2019 20:00:22	Caused by: com.google.api.client.googleapis.json.GoogleJsonResponseException: 403 Forbidden
error	06-Mar-2019 20:00:22	{
error	06-Mar-2019 20:00:22	  "code" : 403,
error	06-Mar-2019 20:00:22	  "errors" : [ {
error	06-Mar-2019 20:00:22	    "domain" : "androidpublisher",
error	06-Mar-2019 20:00:22	    "message" : "APK specifies a version code that has already been used.",
error	06-Mar-2019 20:00:22	    "reason" : "apkUpgradeVersionConflict"
build	06-Mar-2019 20:00:22	33 actionable tasks: 4 executed, 29 up-to-date
error	06-Mar-2019 20:00:22	  } ],
error	06-Mar-2019 20:00:22	  "message" : "APK specifies a version code that has already been used."
error	06-Mar-2019 20:00:22	}
error	06-Mar-2019 20:00:22		at com.google.api.client.googleapis.json.GoogleJsonResponseException.from(GoogleJsonResponseException.java:150)
error	06-Mar-2019 20:00:22		at com.google.api.client.googleapis.services.json.AbstractGoogleJsonClientRequest.newExceptionOnError(AbstractGoogleJsonClientRequest.java:113)
error	06-Mar-2019 20:00:22		at com.google.api.client.googleapis.services.json.AbstractGoogleJsonClientRequest.newExceptionOnError(AbstractGoogleJsonClientRequest.java:40)
error	06-Mar-2019 20:00:22		at com.google.api.client.googleapis.services.AbstractGoogleClientRequest.executeUnparsed(AbstractGoogleClientRequest.java:432)
error	06-Mar-2019 20:00:22		at com.google.api.client.googleapis.services.AbstractGoogleClientRequest.executeUnparsed(AbstractGoogleClientRequest.java:352)
error	06-Mar-2019 20:00:22		at com.google.api.client.googleapis.services.AbstractGoogleClientRequest.execute(AbstractGoogleClientRequest.java:469)
error	06-Mar-2019 20:00:22		at com.github.triplet.gradle.play.tasks.PublishApk.publishApk(PublishApk.kt:63)
error	06-Mar-2019 20:00:22		... 58 more
error	06-Mar-2019 20:00:22	
error	06-Mar-2019 20:00:22	
error	06-Mar-2019 20:00:22	* Get more help at https://help.gradle.org
error	06-Mar-2019 20:00:22	
error	06-Mar-2019 20:00:22	BUILD FAILED in 1m 51s
build	06-Mar-2019 20:00:22	
build	06-Mar-2019 20:00:22	
build	06-Mar-2019 20:00:22	The build scan was not published due to a configuration problem.
build	06-Mar-2019 20:00:22	
build	06-Mar-2019 20:00:22	The Gradle Terms of Service have not been agreed to.
build	06-Mar-2019 20:00:22	
build	06-Mar-2019 20:00:22	To agree to the Gradle Terms of Service, include the following in your root project's configuration:
build	06-Mar-2019 20:00:22	buildScan { termsOfServiceUrl = 'https://gradle.com/terms-of-service'; termsOfServiceAgree = 'yes' }
build	06-Mar-2019 20:00:22	
build	06-Mar-2019 20:00:22	For more information, please see https://gradle.com/scans/help/plugin-terms-of-service.
build	06-Mar-2019 20:00:22	
build	06-Mar-2019 20:00:22	Alternatively, if you are using Gradle Enterprise, specify the server location.
build	06-Mar-2019 20:00:22	For more information, please see https://gradle.com/scans/help/plugin-enterprise-config.
build	06-Mar-2019 20:00:22	
simple	06-Mar-2019 20:00:23	Failing task since return code of [/opt/atlassian/bamboo/temp/runInDocker494432707447173389.sh /var/atlassian/application-data/bamboo/temp/GSDC-GSA-JOB1-16-ScriptBuildTask-97456339034608373.sh] was 1 while expected 0
simple	06-Mar-2019 20:00:23	Finished task 'Build' with result: Failed
simple	06-Mar-2019 20:00:23	Running post build plugin 'NCover Results Collector'
simple	06-Mar-2019 20:00:23	Running post build plugin 'Artifact Copier'
simple	06-Mar-2019 20:00:23	Running post build plugin 'Sauce OnDemand Post Runner'
simple	06-Mar-2019 20:00:23	Running post build plugin 'npm Cache Cleanup'
simple	06-Mar-2019 20:00:23	Running post build plugin 'Clover Results Collector'
simple	06-Mar-2019 20:00:23	Running post build plugin 'Docker Container Cleanup'
simple	06-Mar-2019 20:00:23	Finalising the build...
simple	06-Mar-2019 20:00:23	Stopping timer.
simple	06-Mar-2019 20:00:23	Build GSDC-GSA-JOB1-16 completed.
command	06-Mar-2019 20:00:23	Beginning to execute external process for build 'Groove Studio - Dance Classes - Groove Studio (Android) - Default Job #16 (GSDC-GSA-JOB1-16)'\n ... running command line: \n/usr/bin/docker rm -f 914eb96b-c4ea-4f32-9430-c83bcb14307058327056\n ... in: /var/atlassian/application-data/bamboo/xml-data/build-dir/GSDC-GSA-JOB1\n
build	06-Mar-2019 20:00:24	914eb96b-c4ea-4f32-9430-c83bcb14307058327056
simple	06-Mar-2019 20:00:24	Running on server: post build plugin 'NCover Results Collector'
simple	06-Mar-2019 20:00:24	Running on server: post build plugin 'Build Hanging Detection Configuration'
simple	06-Mar-2019 20:00:24	Running on server: post build plugin 'Clover Delta Calculator'
simple	06-Mar-2019 20:00:24	Running on server: post build plugin 'Maven Dependencies Postprocessor'
simple	06-Mar-2019 20:00:24	All post build plugins have finished
simple	06-Mar-2019 20:00:24	Generating build results summary...
simple	06-Mar-2019 20:00:24	Saving build results to disk...
simple	06-Mar-2019 20:00:24	Logging substituted variables...
simple	06-Mar-2019 20:00:24	Indexing build results...
simple	06-Mar-2019 20:00:24	Finished building GSDC-GSA-JOB1-16.
```

--- a/plugin/src/main/kotlin/com/github/triplet/gradle/play/tasks/ProcessPackageMetadata.kt
+++ b/plugin/src/main/kotlin/com/github/triplet/gradle/play/tasks/ProcessPackageMetadata.kt
@@ -32,10 +32,8 @@ open class ProcessPackageMetadata : PlayPublishTaskBase() {
         val smallestVersionCode = outputs.map { it.versionCode }.min() ?: 1
 
         val patch = maxVersionCode - smallestVersionCode + 1
-        if (patch <= 0) return@read // Nothing to do, outputs are already greater than remote
-
         for ((i, output) in outputs.withIndex()) {
-            output.versionCodeOverride = output.versionCode + patch.toInt() + i
+            if (patch > 0) output.versionCodeOverride = output.versionCode + patch.toInt() + i
             extension._outputProcessor?.execute(output)
         }
     }

